### ISSUE TITLE
pSync still in use semantic

### ISSUE THREAD

**Issue by jdinan** [openshmem-org/specification#181]
In the collectives, we specify the following (example from shmem_alltoall, but the same text appears in all of the collectives):

> The \VAR{pSync} array on all \acp{PE} in the active set is not still in use from a prior call to a \FUNC{shmem\_alltoall/s} routine.

With the addition of ```SHMEM_SYNC_SIZE```, a given ```pSync``` array can be used by any collective operation (e.g. broadcast, alltoall, etc.).  Thus, this text should be updated to be a bit more general:

> The \VAR{pSync} array on all \acp{PE} in the active set is not still in use from a prior operation.

**Comment by manjugv** [comment]
This seems like a semantic change. With this change, we have to ensure that a pSync used for the collective operation is not used by any collective operation rather than just the same type of collective operation as currently specified. 

May be the text should be more specific:

> The pSync array on all PEs in the active set is not still in use from a prior call to a shmem_alltoall routine, if the pSync array used is specific to the shmem_alltoall operation and size SHMEM_PSYNC_ALLTOALL. Otherwise, if the same pSync array of size SHMEM_PSYNC_SIZE is used for all collective operations, the pSync array should not be in use from a prior collective operation.


Though this is a correct clarification to make, I will keep this issue open and change the target to 1.5. I don’t want to push this now.

**Comment by jdinan** [comment]
I suppose we can save this for 1.5 if you're concerned that the semantics should be hashed out using a more deliberative process; however, if you do feel that the semantic direction is clear (and have consensus from the rest of the section committee) it's fair game to clean this up in the committee since SHMEM_SYNC_SIZE is being introduced in 1.4.

The concern I have with letting this one slip to 1.5 is that one could read between the lines and conclude that if you use a SHMEM_SYNC_SIZE psync array, then it's ok to call alltoall followed by reduction without an intervening barrier.  That's certainly not what we intended when we added SHMEM_SYNC_SIZE.
